test/common/upstream/cds_api_impl_test.cc

```diff
@@ -81,7 +81,10 @@ class CdsApiImplTest : public testing::Test {
                             {":method", v2_rest_ ? "POST" : "GET"},
                             {":path", v2_rest_ ? "/v2/discovery:clusters"
                                                : "/v1/clusters/cluster_name/node_name"},
-                            {":authority", "foo_cluster"}}),
+                            {":authority", "foo_cluster"},
+                            {"content-type", "application/json"},
+                            {"content-length",
+                             v2_rest_ ? fmt::FormatInt(request->body()->length()).str() : "0"}}),
```

Comment:
Do we need to pivot on v2 here? Can we just pivot on the existence of a body or not? Same below.